### PR TITLE
Added CREATED status for new vulnerability

### PR DESCRIPTION
**Pull request by Cheshiriks**
- Added CREATED status for new vulnerability

### CODE REVIEW

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt [hunk 1]
```diff
@@ -20,6 +20,11 @@ enum class VulnerabilityStatus {
      */
     AUTO_APPROVED,
 
+    /**
+     * Created status
+     */
+    CREATED,
+
```
Comment (DrAlexD) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt:27]: Is it clear enough for users and admins that `CREATED` status also requires approve from admins, just like `PENDING_REVIEW`? Is status `CREATED` necessary? Maybe we should rename it somehow and then maybe also `PENDING_REVIEW` status?

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/VulnerabilityMetadataService.kt [hunk 2]
```diff
@@ -172,6 +171,12 @@ class VulnerabilityMetadataService(
...
                 } ?: 0f
 
-        private fun Boolean.toVulnerabilityStatus() = if (this) VulnerabilityStatus.AUTO_APPROVED else VulnerabilityStatus.PENDING_REVIEW
+        private fun Boolean.toVulnerabilityStatus(isNew: Boolean) = if (this) {
+            VulnerabilityStatus.AUTO_APPROVED
+        } else if (isNew) {
+            VulnerabilityStatus.CREATED
+        } else {
+            VulnerabilityStatus.PENDING_REVIEW
+        }
```
Comment (DrAlexD) [inline save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/VulnerabilityMetadataService.kt:180]: Maybe change this function to something like `private fun getVulnerabilityStatus(isAutoApprove: Boolean, isNew: Boolean)`?

**Review by kgevorkyan — APPROVED** [review]

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt:27]: Statuses are hidden in the system. There are only two states for users: _Approved_ and _Not Approved_.
But it is important for us to distinguish new vulnerabilities from those that have simply been changed.
Changing new ones will not have to create new versions. While old vulnerabilities must have a copy of the version before and after the change.

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/VulnerabilityMetadataService.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/VulnerabilityMetadataService.kt]: Ok, changed it

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt (hunk 1 above)
Comment (DrAlexD) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt:27]: Ok, agreed

**Review by nulls — APPROVED** [review]

**Pull request merged**